ledger/src/blockstore.rs

```diff
@@ -2031,6 +2031,33 @@ impl Blockstore {
         slot: Slot,
         require_previous_blockhash: bool,
     ) -> Result<VersionedConfirmedBlock> {
+        self.get_complete_block_with_entries(slot, require_previous_blockhash, false)
+            .map(|result| result.block)
+    }
+
+    pub fn get_rooted_block_with_entries(
+        &self,
+        slot: Slot,
+        require_previous_blockhash: bool,
+    ) -> Result<VersionedConfirmedBlockWithEntries> {
+        datapoint_info!(
```

Comment:
Will this method get bubbled up to a higher level API (ie RPC) as well? `get_rooted_block()` will emit two datapoints; `get_rooted_block` and `get_rooted_block_with_entries` for the subfunction. This isn't necessarily a problem, but something I had noticed and been thinking about a little when looking through the blockstore code.

Some additional context for my question - part of me moving the blockstore metric service over to ledger crate in https://github.com/solana-labs/solana/pull/33947 was so that I could condense all of the `blockstore-rpc-api` datapoints into a struct + single datapoint (something similar to https://github.com/solana-labs/solana/pull/31691). For example, there is something similar in that `get_rooted_transaction_status()` calls `get_transaction_status()`, both of which emit a datapoint. Maybe the "double count" doesn't matter so much if these are just counters as opposed to datapoints with strings.

This is a minor point for this PR, but mentioning it because I saw it. I can always sweep this up later in the work I mentioned that I plan to do

Reply:
Currently, I only plan to call this method from things that upload to bigtable (ledger-tool and bigtable_upload_service).
`get_complete_block_with_entries()` doesn't actually emit a datapoint, though, and my intent was to leave it a private function so that it doesn't need one. So this doesn't add any doubles. I could leave this out altogether, though, to avoid adding another string datapoint. I don't feel very strongly about having it.

Interesting that `get_complete_block()` hasn't ever emitted a metric, though. I can't remember if I was explicitly trying to avoid the double report or just didn't think of it...

Reply:
> get_complete_block_with_entries() doesn't actually emit a datapoint, though, and my intent was to leave it a private function so that it doesn't need one. So this doesn't add any doubles.

D'oh, you're right. The two call paths are:
- `get_rooted_block() ==> get_complete_block() ==> get_complete_block_with_entries()` AND
- `get_rooted_block_with_entries() ==> get_complete_block_with_entries()`

I mistook `get_rooted_block()` as calling `get_rooted_block_with_entries()`